camel/embeddings/openai_embedding.py

```diff
@@ -0,0 +1,88 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import os
+from enum import Enum
+from typing import List
+
+from openai import OpenAI
+
+from camel.embeddings import BaseEmbedding
+
+
+class OpenAIEmbeddingModel(Enum):
+    ADA2 = "text-embedding-ada-002"
+    ADA1 = "text-embedding-ada-001"
+    BABBAGE1 = "text-embedding-babbage-001"
+    CURIE1 = "text-embedding-curie-001"
+    DAVINCI1 = "text-embedding-davinci-001"
```

Comment:
I prefer to put ADA_2 first. It's the only 2nd generation model. Other old (possibly never used) model should be put together.

Reply:
Sure, maybe let's name it as `EmbeddingModelType ` since we will have other EmbeddingModel to be added. just like `ModelType`. Moreover, should we move it to `types`?